test/python/unittest/API/APITest.py

```diff
@@ -0,0 +1,339 @@
+        """ Test querying a particle species. """
+
+        # Get series.
+        series = self.__series
+
+        # Check type.
+        i = series.iterations[100]
+
+        self.assertTrue(hasattr(i.particles, "__getitem__"))
+        for ps in i.particles:
+            self.assertIsInstance(ps, str)
+            self.assertIsInstance(i.particles[ps], openPMD.ParticleSpecies)
+
+    def testAllocation(self):
+        """ Test openPMD.Allocation. """
+        obj = openPMD.Allocation(1)
+        del obj
+
+    def testData_Order(self):
+        """ Test openPMD.Data_Order. """
+        obj = openPMD.Data_Order('C')
+        del obj
+
+    def testDatatype(self):
+        """ Test openPMD.Datatype. """
+        data_type = openPMD.Datatype(1)
+        del data_type
+
+    def testDataset(self):
+        """ Test openPMD.Dataset. """
+        data_type = openPMD.Datatype(1)
+        extent = openPMD.Extent()
+        obj = openPMD.Dataset(data_type, extent)
+        del obj
+
+    def testExtent(self):
+        """ Test openPMD.Extent. """
+        obj = openPMD.Extent()
+        del obj
+
+    def testGeometry(self):
+        """ Test openPMD.Geometry. """
+        obj = openPMD.Geometry(0)
+        del obj
+
+    def testIteration(self):
+        """ Test openPMD.Iteration. """
+        self.assertRaises(TypeError, openPMD.Iteration)
+
+        iteration = self.__particle_series.iterations[400]
+
+        copy_iteration = openPMD.Iteration(iteration)
+
+        self.assertIsInstance(copy_iteration, openPMD.Iteration)
+
+    def testIteration_Container(self):
+        """ Test openPMD.Iteration_Container. """
+        obj = openPMD.Iteration_Container()
+        del obj
+
+    def testIteration_Encoding(self):
+        """ Test openPMD.Iteration_Encoding. """
+        obj = openPMD.Iteration_Encoding(1)
+        del obj
+
+    def testMesh(self):
+        """ Test openPMD.Mesh. """
+        self.assertRaises(TypeError, openPMD.Mesh)
+        mesh = self.__series.iterations[100].meshes['E']
+        copy_mesh = openPMD.Mesh(mesh)
+
+        self.assertIsInstance(copy_mesh, openPMD.Mesh)
+
+    def testMesh_Container(self):
+        """ Test openPMD.Mesh_Container. """
+        obj = openPMD.Mesh_Container()
+        del obj
+
+    def testParticlePatches(self):
+        """ Test openPMD.ParticlePatches. """
+        self.assertRaises(TypeError, openPMD.ParticlePatches)
+
+    def testParticleSpecies(self):
+        """ Test openPMD.ParticleSpecies. """
+        self.assertRaises(TypeError, openPMD.ParticleSpecies)
+
+    def testParticle_Container(self):
+        """ Test openPMD.Particle_Container. """
+        obj = openPMD.Particle_Container()
+        del obj
+
+    def testRecord(self):
+        """ Test openPMD.Record. """
+        # Has only copy constructor.
+        self.assertRaises(TypeError, openPMD.Record)
+#        ### FIXME
+#        ## Get a record (fails)
+#        #record = self.__series.iterations[100].particles[
```

Comment:
can you try if

```python
record = self.__series.iterations[100].particles['electrons']['positions']['x']
```
works?

Reply:
ERROR: testRecord (__main__.APITest)
Test openPMD.Record.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "APITest.py", line 293, in testRecord
    record = self.__series.iterations[100].particles['electrons']['positions']['x']
TypeError: 'openPMD.ParticleSpecies' object is not subscriptable

Reply:
Ah, alright. Not implemented yet, a `FIXME` note is great.